src/main.rs

```diff
@@ -490,7 +484,7 @@ impl Config {
         }
 
         let repo_access: Box<dyn RustRepositoryAccessor>;
-        repo_access = match args.access.as_ref().map(|x| x.as_str()) {
+        repo_access = match args.access.as_deref() {
```

Comment:
warning: called `.as_ref().map(|x| x.as_str())` on an Option value. This can be done more directly by calling `args.access.as_deref()` instead